starter-code/TicTacToe/app/src/main/java/ly/generalassemb/drewmahrt/tictactoe/GameActivity.java

```diff
@@ -1,13 +1,234 @@
+
+    public void whoseTurn() {
+        if (mTurn % 2 == 0) {
+            mLetter = "X";
+            mGameMessage.setText(mUsername2 + "'s Turn");
+        } else {
+            mLetter = "O";
+            mGameMessage.setText(mUserName1 +"'s Turn");
+        }
+    }
+
+    public void resultWin() {
+        if (win == true) {
+            if (mTurn % 2 == 0) {
+                mGameMessage.setText(mUserName1 +" WON!");
+                mWinner = mUserName1;
+                mBackButton.setVisibility(View.VISIBLE);
+            } else {
+                mGameMessage.setText(mUsername2 +" WON!");
+                mWinner = mUsername2;
+                mBackButton.setVisibility(View.VISIBLE);
+            }
+        } else if (mTurn == 10){
+            mGameMessage.setText("Round Tie!");
+            mWinner = "Round Tie!";
+            mBackButton.setVisibility(View.VISIBLE);
+        }
+    }
+
+    CountDownTimer timer = new CountDownTimer(1, 1) {
+        @Override
+        public void onTick(long l) {
+
+        }
+
+        @Override
+        public void onFinish() {
+            winCondition();
+            resultWin();
+        }
+    };
+
+    public void winCondition() {
+        //horizontalwins
+        if (mTextView1.getText() == mTextView2.getText() &&
+                mTextView2.getText() == mTextView3.getText() && mTextView1.getText() != "") {
+            win = true;
+        } else if (mTextView4.getText() == mTextView5.getText() &&
+                mTextView5.getText() == mTextView6.getText() && mTextView4.getText() != "") {
+            win = true;
+        } else if (mTextView7.getText() == mTextView8.getText() &&
+                mTextView8.getText() == mTextView9.getText() && mTextView7.getText() != "") {
+            win = true;
+        }
+
+        //verticle wins
+        else if (mTextView1.getText() == mTextView4.getText() &&
+                mTextView4.getText() == mTextView7.getText() && mTextView1.getText() != "") {
+            win = true;
+        } else if (mTextView2.getText() == mTextView5.getText() &&
+                mTextView5.getText() == mTextView8.getText() && mTextView2.getText() != "") {
+            win = true;
+        } else if (mTextView3.getText() == mTextView6.getText() &&
+                mTextView6.getText() == mTextView9.getText() && mTextView3.getText() != "") {
+            win = true;
+        }
+
+        //diagonal wins
+        else if (mTextView1.getText() == mTextView5.getText() &&
+                mTextView5.getText() == mTextView9.getText() && mTextView1.getText() != "") {
+            win = true;
+        } else if (mTextView3.getText() == mTextView5.getText() &&
+                mTextView5.getText() == mTextView7.getText() && mTextView3.getText() != "") {
+            win = true;
+        } else {
+            win = false;
+        }
+    }
 
-public class GameActivity extends AppCompatActivity {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
```

Comment:
Best practice is usually to put `onCreate()` and other overridden methods from Activity at the top, right after your member variables, then put your helper methods below.